### PR TITLE
Fix #1646: Implement Character.is{Upper,Lower}Case as explicit regex.

### LINKED ISSUE #1646: Char#isUpper behavior diverges between Scala.js/Scala-JVM
```scala
scala> ')'.isUpper
false

scala-js> ')'.isUpper
true
```

`Char#isLower` also shares this behavior, discovered porting https://github.com/lihaoyi/fastparse to Scala.js

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd.

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/CharacterTest.scala [hunk 1]
```diff
@@ -679,5 +679,438 @@ object CharacterTest extends JasmineTest {
       expect(Character.isUnicodeIdentifierPart('\u2d2c')).toBeFalsy
     }
 
+    it("should provide isUpper") {
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/CharacterTest.scala]: isUpper<ins>Case</ins>

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/CharacterTest.scala [hunk 2]
```diff
@@ -679,5 +679,438 @@ object CharacterTest extends JasmineTest {
...
+      expect(Character.isUpperCase('\u8473')).toBeFalsy
+      expect(Character.isUpperCase('\u478B')).toBeFalsy
+      expect(Character.isUpperCase('\uB916')).toBeFalsy
+      expect(Character.isUpperCase('\uBBFA')).toBeFalsy
+      expect(Character.isUpperCase('\uDC65')).toBeFalsy
+      expect(Character.isUpperCase('\u65E9')).toBeFalsy
+      expect(Character.isUpperCase('\u97FF')).toBeFalsy
+    }
+
+    it("should provide isLower") {
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/CharacterTest.scala]: isLower<ins>Case</ins>

**Comment by sjrd** [comment]
The commit title is a bit too long (you can see it wrap in GitHub). You can shorten it by writing `Character.is{Upper,Lower}Case`.

**Comment by sjrd** [comment]
That's all.

**Comment by nicolasstucki** [comment]
Updated.

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
retest this please

**Comment by sjrd** [comment]
It seems this PR breaks checksizes.sh.

File: javalanglib/src/main/scala/java/lang/Character.scala [hunk 3]
```diff
@@ -303,6 +308,13 @@ object Character {
...
+
+
```
Comment (sjrd) [inline javalanglib/src/main/scala/java/lang/Character.scala]: And just noticed this now: Double blank line.

**Comment by lihaoyi** [comment]
lol, and here I was generating the upper/lower cases using a [macro pre-computed bitset](https://github.com/lihaoyi/fastparse/blob/master/utils/shared/src/main/scala/fastparse/Utils.scala#L32-L37)

Will be so glad to start ripping those out when this lands and I can shrink the size of scalaparse by dozens of KB. Hopefully some day you'll be able to implement [the rest of the character predicates...](https://github.com/lihaoyi/fastparse/blob/master/fastparse/js/src/main/scala/fastparse/CharPredicates.scala)

FWIW pre-computing the bitset in JVM-land and looping over all 65536 characters could be a good way to verify the "this should behave the same as the JVM" invariant. Otherwise, even looking at the test suite, I have no idea whether it's verifying the right things or not...

Including the macro pre-computer and for-looping over everything may even be fewer lines of code than the manual asserts

**Comment by nicolasstucki** [comment]
@lihaoyi, I grant you that the regex might not be the most compact representation. Our first goal is to solve the bug, in the we could worry about performance.

To generate this regex and the tests I actually looped through all the characters in the JVM and checked the values returned by the functions.

**Comment by nicolasstucki** [comment]
Updated `checksizes.sh`

**Comment by sjrd** [comment]
No change for 2.12.0-M1?

**Comment by nicolasstucki** [comment]
Maybe there was a change, but I did not see a `checksizes.sh` failures on that version.

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
Here's your failing build for 2.12.0-M1: https://scala-webapps.epfl.ch/jenkins/job/scalajs-task-worker/46987/console ^^

**Comment by sjrd** [comment]
And actually here 2.11.6 seems to have a worse version: https://scala-webapps.epfl.ch/jenkins/job/scalajs-task-worker/46984/console (can sometimes depend on the JDK)

**Comment by nicolasstucki** [comment]
Updated

**Pull request merged**